### PR TITLE
Gradle upgrade 5.6.4

### PR DESCRIPTION
**Pull request by shy950521**
ext.commonBuildVersion = '5.6.4'  not released yet;

add compile log4j:log4j:1.2.17 in dependency, the runtime included by org.slf4j:slf4j-api:1.7.5 seems not working.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -24,22 +24,264 @@
...
+    apply plugin: 'eclipse'
+    apply plugin: 'java'
+    apply plugin: 'net.saliman.cobertura'
+    apply plugin: 'distribution'
+    apply plugin: 'signing'
+    apply plugin: 'maven'
+
+    defaultTasks 'distZip'
+
+    def isRootProject = (project == rootProject)
```
Comment (twincitiesguy) [inline build.gradle:60]: The rootProject variable is also a common-build construct.  We know this is the root project, so there is no need for this variable.

File: build.gradle [hunk 2]
```diff
@@ -24,22 +24,264 @@
...
+            if (!rootProject.hasProperty('signingPass')) {
+                rootProject.ext.signingPass = new String(System.console().readPassword('\nSigning key passphrase: '))
+                rootProject.ext.sonatypePass = new String(System.console().readPassword('\nSonatype password: '))
+            }
+            ext.'signing.keyId' = signingKeyId
+            ext.'signing.secretKeyRingFile' = signingSecretKeyRingFile
+            ext.'signing.password' = rootProject.ext.signingPass
+            uploadJars.repositories.mavenDeployer.repository.authentication.password = rootProject.ext.sonatypePass
+        }
+        if (isRootProject) {
```
Comment (twincitiesguy) [inline build.gradle:79]: We know this is the root project, so don’t need this conditional.

File: build.gradle [hunk 3]
```diff
@@ -24,22 +24,264 @@
...
+                System.setProperty('org.ajoberstar.grgit.auth.username', gitUsername)
+                System.setProperty('org.ajoberstar.grgit.auth.password', gitPassword)
+            }
+        }
+    }
+
+    [compileJava, compileTestJava]*.options*.encoding = 'UTF-8'
+
+    compileJava {
+        sourceCompatibility = 1.6
```
Comment (twincitiesguy) [inline build.gradle:94]: Bump these to 1.8

File: build.gradle [hunk 4]
```diff
@@ -24,22 +24,264 @@
...
+        targetCompatibility = 1.6
+        if (project.hasProperty('java6Lib')) {
```
Comment (twincitiesguy) [inline build.gradle:96]: Change to java8Lib

File: build.gradle [hunk 5]
```diff
@@ -24,22 +24,264 @@
...
+            ).join(File.pathSeparator)
+        }
+    }
+
+    repositories {
+        mavenCentral()
+        mavenLocal()
+    }
+
+    test {
```
Comment (twincitiesguy) [inline build.gradle:109]: Don’t need this with Java 8

File: build.gradle [hunk 6]
```diff
@@ -24,22 +24,264 @@
...
+                    name licenseName
+                    url licenseUrl
+                    distribution 'repo'
+                }
+            }
+
+            developers {
+                developer {
+                    id 'EMCECS'
+                    name 'EMC ViPR'
```
Comment (twincitiesguy) [inline build.gradle:136]: Update Pom references to Dell and ECS branding

File: build.gradle [hunk 7]
```diff
@@ -24,22 +24,264 @@
...
+                        'Class-Path': configurations.runtime.collect { it.getName() }.join(' ')
+            }
+        }
+        into("META-INF/maven/$project.group/$project.name") {
+            from writePom
+        }
+    }
+
+    javadoc {
+        if (JavaVersion.current() >= JavaVersion.VERSION_1_8) options.addStringOption('Xdoclint:none', '-quiet')
```
Comment (twincitiesguy) [inline build.gradle:163]: Assume this will always be Java 8+

File: build.gradle [hunk 8]
```diff
@@ -24,22 +24,264 @@
...
+                from docsDir
+                into "${rootProject.aggregatedDocsDir}/${project.version}${docSubDir}"
+            }
+        }
+    }
+    aggregateDocs.dependsOn javadoc
+    aggregateDocs.dependsOn { subprojects.collect { it.tasks['aggregateDocs'] } }
+
+// we can't release or publish sub-projects
+    if (isRootProject) {
```
Comment (twincitiesguy) [inline build.gradle:246]: We know this is the root project

File: build.gradle [hunk 9]
```diff
@@ -24,22 +24,264 @@
...
-buildscript {
-    ext.commonBuildVersion = '1.6'
-    ext.commonBuildDir = "https://raw.githubusercontent.com/EMCECS/ecs-common-build/v$commonBuildVersion"
-    apply from: "$commonBuildDir/ecs-publish.buildscript.gradle", to: buildscript
-}
-
 allprojects {
-    apply from: "$commonBuildDir/ecs-publish.gradle"
+    group 'com.emc.ecs'
+
```
Comment (twincitiesguy) [inline build.gradle:40]: Move the plug-in block (below) to here and then close the allProjects block.. I don’t think any of the rest of this block is necessary for the geopinning tool.

Comment (twincitiesguy) [reply build.gradle:40]: Might also need to move the tools configuration here too

**Review by twincitiesguy — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 9 above)
Comment (shy950521) [reply build.gradle:40]: allproject now only contains: param setting, plugin apply block, configurations and repositories

**Review by twincitiesguy — APPROVED** [review]

**Pull request merged**